build_tools/github_actions/should_ci_run.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Determines whether CI should run on a given PR.
+
+Exit code 0 indicates that it should and exit code 2 indicates that it should
+not.
+"""
+
+import fnmatch
+import os
+import subprocess
+import sys
+
+SKIP_CI_LABEL = "skip-ci"
+
+# Note that these are fnmatch patterns, which are not the same as gitignore
+# patterns because they don't treat '/' specially. The standard library doesn't
+# contain a function for gitignore style "wildmatch". There's a third-party
+# library pathspec (https://pypi.org/project/pathspec/), but it doesn't seem
+# worth the dependency.
+SKIP_PATH_PATTERNS = [
+    "docs/*",
+    "experimental/*",
+    "build_tools/kokoro/*",
+    "build_tools/buildkite/*",
+    ".github/ISSUE_TEMPLATE/*",
+    "*.cff",
+    "*.clang-format",
+    "*.git-ignore",
+    "*.md",
+    "*.natvis",
+    "*.pylintrc",
+    "*.rst",
+    "*.toml",
+    "*.yamllint.yml",
+    "*.yapf",
+    "*CODEOWNERS",
+    "*AUTHORS",
+    "*LICENSE",
+]
+
+
+def skip_path(path):
+  return any(fnmatch.fnmatch(path, pattern) for pattern in SKIP_PATH_PATTERNS)
+
+
+def get_modified_paths(base_ref):
+  return subprocess.run(["git", "diff", "--name-only", base_ref],
```

Comment:
How about using subprocess.check_output(...).splitlines() ?

Reply:
`check_output` is the old api. `run` is recommended now (see https://docs.python.org/3/library/subprocess.html)